libbeat/reader/split/splitter.go

```diff
@@ -0,0 +1,214 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package split
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+var (
+	errEmptyField       = errors.New("the requested field is empty")
+	errEmptyRootField   = errors.New("the requested root field is empty")
+	errExpectedSplitArr = errors.New("split was expecting field to be an array")
+	errExpectedSplitObj = errors.New("split was expecting field to be an object")
+)
+
+// split is a split processor chain element. Split processing is executed
+// by applying elements of the chain's linked list to an input until completed
+// or an error state is encountered.
+type split struct {
+	log    *logp.Logger
+	target string
+	// kind   string
+	child            *split
+	keepParent       bool
+	ignoreEmptyValue bool
+	// keyField         string
+	IsRoot bool
+	// delimiter        string
+}
+type maybeMsg struct {
+	err error
+	Msg mapstr.M
+}
+
+func (e maybeMsg) Failed() bool { return e.err != nil }
+
+func (e maybeMsg) Error() string { return e.err.Error() }
+
+// newSplit returns a new split based on the provided config and
+// logging to the provided logger.
+func NewSplit(c *Config, log *logp.Logger) (*split, error) {
+	var s *split
+	if c.Split != nil {
+		s, _ = NewSplit(c.Split, log)
+		// if err != nil {
+		// 	return nil, err
+		// }
+	}
+
+	return &split{
+		log:    log,
+		target: c.Target,
+		// kind:             c.Type,
+		keepParent:       c.KeepParent,
+		ignoreEmptyValue: c.IgnoreEmptyValue,
+		// keyField:         c.KeyField,
+		// delimiter:        c.DelimiterString,
+		child: s,
+	}, nil
+}
+
+func (s *split) StartSplit(raw json.RawMessage) (<-chan maybeMsg, error) {
+	ch := make(chan maybeMsg)
+	var jsonObject mapstr.M
+	if err := json.Unmarshal(raw, &jsonObject); err != nil {
+		return nil, err
+	}
```

Comment:
Here you're trying to decode the message as a JSON object (aka a map), but the example @kvch posted here: https://github.com/elastic/beats/pull/29951#issuecomment-1156594940 uses an array, so the decoding fails.

Also if the input file does not contain an empty line at the end, Filebeat just hangs.